### PR TITLE
#304 Implement pagination

### PR DESCRIPTION
**Pull request by aleksandarmicev**
Implement pagination

### CODE REVIEW

File: components/reusable-components/pagination/HomePagination.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  const [items, setItems] = useState<Item[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(25);
+
+  const options = [25, 50, 100];
+
+  useEffect(() => {
+    const loadItems = async () => {
+      const data = await fetchItems(currentPage, itemsPerPage);
```
Comment (edichoska) [inline components/reusable-components/pagination/HomePagination.tsx:17]: try catch here

File: components/reusable-components/pagination/StructurePagination.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (currentPage < totalPages) {
+      onPageChange(currentPage + 1);
+    }
+  };
```
Comment (edichoska) [inline components/reusable-components/pagination/StructurePagination.tsx:25]: this should be in another PR. Also we should have pages, like 1,2,3,4,5 not next and previous

File: components/reusable-components/pagination/HomePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import Dropdown from './Dropdown';
+import styles from './HomePagination.module.scss';
+
+const HomePagination: React.FC = () => {
+  const [items, setItems] = useState<Item[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalPages, setTotalPages] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(25);
+
+  const options = [25, 50, 100];
```
Comment (edichoska) [inline components/reusable-components/pagination/HomePagination.tsx:13]: the name should be all uppercase letter, it is a convention, also use more descriptive name

File: components/reusable-components/pagination/HomePagination.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      setItems(data.items);
+      setTotalPages(Math.ceil(data.total / itemsPerPage));
+    };
+
+    loadItems();
+  }, [currentPage, itemsPerPage]);
+
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [itemsPerPage]);
```
Comment (edichoska) [inline components/reusable-components/pagination/HomePagination.tsx:27]: consider the following scenario here:
we have 100 users, we are showing 25 per page, user is on page 4, and changes in the dropdown to 50 per page. So now we only have 2 pages, and not 4.
How can we handle that scenario?

**Comment by edichoska** [comment]
@aleksandarmicev Can you fix this comments when you have the time, so we can merge this?
Also resolve the conflict with the project Section

File: components/reusable-components/pagination/HomePagination.tsx [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
 import styles from './HomePagination.module.scss';
 
+/* eslint no-unused-vars: "off" */
 const HomePagination: React.FC = () => {
   const [items, setItems] = useState<Item[]>([]);
   const [currentPage, setCurrentPage] = useState(1);
-  const [totalPages, setTotalPages] = useState(1);
-  const [itemsPerPage, setItemsPerPage] = useState(25);
+  const [TOTAL_PAGES, SET_TOTAL_PAGES] = useState(1);
+  const [ITEMS_PER_PAGE, SET_ITEMS_PER_PAGE] = useState(25);
```
Comment (edichoska) [inline components/reusable-components/pagination/HomePagination.tsx:11]: naming const with capital letters doesnt apply to useState, please revert this changes

File: components/reusable-components/pagination/HomePagination.tsx (hunk 4 above)
Comment (edichoska) [reply components/reusable-components/pagination/HomePagination.tsx]: @aleksandarmicev also please consider the scenario in the comment above. Please tag me when this is done

**Review by edichoska — CHANGES_REQUESTED** [review]

**Comment by edichoska** [comment]
@aleksandarmicev resolve conflicts here

**Comment by aleksandarmicev** [comment]
> @aleksandarmicev resolve conflicts here

Conflicts are just resolved, merged new changes from the staging

File: components/reusable-components/pagination/Dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+      {options.map((option) => (
+        <option key={option} value={option}>
+          {option}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+export default Dropdown;
```
Comment (edichoska) [inline components/reusable-components/pagination/Dropdown.tsx:23]: Don't we already have a reusable dropdown component?

**Comment by edichoska** [comment]
@aleksandarmicev Please use this in combination with the table and some dummy data to test if this is working, and attach a video. 🙏

**Comment by edichoska** [comment]
@aleksandarmicev Is this tested? Could you maybe add it, in the reusable table implementation in content-panel > blogs?

File: components/reusable-components/pagination/HomePagination.tsx
Comment (edichoska) [inline components/reusable-components/pagination/HomePagination.tsx:1]: Maybe a better approach here would be to create a context. Here you should also consider that we will have a page pagination. So we will need some props in multiple components.
Also we should persist the choice the user will make when choosing an option from the select dropdown.

**Review by edichoska — CHANGES_REQUESTED** [review]

**Review by edichoska — CHANGES_REQUESTED** [review]
Please remove comments, commented out imports. Also please remove the inline styles into classes.

**Comment by Tikkurilla** [comment]
Question about the requirements:

The pagination requirements look partial.  Is there any other ticket about actually displaying the pages, total count, current page ? This looks like just defining how many items will be visible per page..or am I missing something

**Comment by edichoska** [comment]
> @edichoska Question about the requirements:
> 
> The pagination requirements look partial. Is there any other ticket about actually displaying the pages, total count, current page ? This looks like just defining how many items will be visible per page..or am I missing something

Yes there was another ticket for that, already done: https://github.com/learnhubmk/app/pull/451

**Comment by Tikkurilla** [comment]
@aleksandarmicev 

We need to close this PR and try to re-use the code to merge this logic under Pagination.tsx component whhic will be genetic component re-suable in all places

**Comment by edichoska** [comment]
Closing this PR since we will update the Pagination.tsx component and that one will be reused everywhere

**Pull request closed without merging**